src/store/actions/userSession.actions.js

```diff
@@ -115,20 +144,25 @@ export const logout = () => dispatch => {
 export const getUserStatus = () => dispatch => {
   dispatch(checkLoggedStatus())
   const JWTtoken = getAccessToken()
+  const username = getUserNameFromToken(JWTtoken) || getUserName()
+  const userrole = getUserRoleFromToken(JWTtoken) || true
```

Comment:
Are there any specific reasons to not be camelCased?

Reply:
And why we are creating variables at all?
If `JWTtoken` is **false**, we won't need `username` and `userrole`.

Reply:
Token obtained by logging in with username and password method does not contain username and user role in its payload.

What is done here is to store the username written in the form in localStorage and assume the user is admin.

Reply:
Yeap, that is pretty obvious.

```js
if (JWTtoken) {
  dispatch(loginSuccess(
    // here you are getting 'name'
    ,// and here - 'role'
  )
} else {
...
```

Reply:
if `JWTtoken` is false - previously created variables are senseless.

Reply:
Note that `JWTtoken` is not false nor empty with the basic authentication.
It is `getUserNameFromToken()` and `getUserRoleFromToken()` that fails if run against token obtained from Janus directly (and not github).

Reply:
The two variables are created only for readability.

Reply:
if it is not empty - then we do not need the "if" part at all, no?

Reply:
It's still needed (in case the user is not logged in), but you are right in that both variables should be moved inside the condition block 👍 . It does not make sense to get the username and role when there is no token.

Reply:
exactly my point 😄 👍